File: SEMESTER 2/Java/Cycle 3/Inheritance.java
class Grandparent
{
    String gname;
    int gage;

    public Grandparent(String gname, int gage)
    {
        this.gname = gname;
        this.gage = gage;
    }

    public void grandparentDetails()
    {
        System.out.println("\nGrandparent: ");
        System.out.println("Name: " + gname);
        System.out.println("Age: " + gage);
    }
}

class Parent extends Grandparent
{
    String pname;
    int page;

    public Parent(String gname, int gage, String pname, int page)
    {
        super(gname, gage);
        this.pname = pname;
        this.page = page;
    }

    public void parentDetails()
    {
        super.grandparentDetails();
        System.out.println("\nParent: ");
        System.out.println("Name: " + pname);
        System.out.println("Age: " + page);
    }
}

class Child extends Parent 
{
    String cname;
    int cage;
    
    public Child(String gname, int gage, String pname, int page, String cname, int cage) 
    {
        super(gname, gage, pname, page);
        this.cname = cname;
        this.cage = cage;
    }
    
    public void childDetails() 
    {
        super.parentDetails();
        System.out.println("\nChild: ");
        System.out.println("Name: " + cname);
        System.out.println("Age: " + cage);
    }
}

public class Inheritance
{
    public static void main(String[] args) 
    {
        Child child = new Child("Grandpa", 70, "Dad", 40, "Child", 10);
        child.childDetails();
    }
}

File: SEMESTER 3/PYTHON/Cycle 2/inheritance.py
# Define the Grandparent class
class Grandparent:
    def __init__(self, gname, gage):
        self.gname = gname
        self.gage = gage

    def GrandparentDetails(self):
        print("Grandparent Name:", self.gname)
        print("Grandparent Age:", self.gage)

# Define the Parent class that inherits from Grandparent
class Parent(Grandparent):
    def __init__(self, gname, gage, pname, page):
        # Initialize properties of the Grandparent class
        super().__init__(gname, gage)
        self.pname = pname
        self.page = page

    def ParentDetails(self):
        print("Parent Name:", self.pname)
        print("Parent Age:", self.page)

# Define the Child class that inherits from Parent
class Child(Parent):
    def __init__(self, gname, gage, pname, page, cname, cage):
        # Initialize properties of the Parent class (and Grandparent)
        super().__init__(gname, gage, pname, page)
        self.cname = cname
        self.cage = cage

    def ChildDetails(self):
        print("Child Name:", self.cname)
        print("Child Age:", self.cage)

# Create an object for Child and display all details
child1 = Child("Robert", 78, "Michael", 50, "Emma", 20)

# Display the details of each family member
child1.GrandparentDetails()
child1.ParentDetails()
child1.ChildDetails()
